apps/web/src/components/layout/components/v2/NewDashboardOptInWidget.tsx

```diff
@@ -0,0 +1,112 @@
+import { Card } from '@mantine/core';
+import { css } from '@novu/novui/css';
+import { Text, Title, Button, IconButton } from '@novu/novui';
+import { IconOutlineClose } from '@novu/novui/icons';
+import { useUser } from '@clerk/clerk-react';
+import { FeatureFlagsKeysEnum, NewDashboardOptInStatusEnum } from '@novu/shared';
+import { IS_SELF_HOSTED } from '../../../../config';
+import { useFeatureFlag } from '../../../../hooks';
+
+export function NewDashboardOptInWidget() {
+  const { user } = useUser();
+  const isNewDashboardEnabled = useFeatureFlag(FeatureFlagsKeysEnum.IS_NEW_DASHBOARD_ENABLED);
+
+  const isDismissed = user?.unsafeMetadata?.newDashboardOptInStatus === NewDashboardOptInStatusEnum.DISMISSED;
+
+  if (IS_SELF_HOSTED || isDismissed || !isNewDashboardEnabled) {
+    return null;
+  }
+
+  const updateUserOptInStatus = (status: NewDashboardOptInStatusEnum) => {
+    if (!user) return;
+
+    user.update({
+      unsafeMetadata: {
+        ...user.unsafeMetadata,
+        newDashboardOptInStatus: status,
+      },
+    });
+  };
+
+  function handleOptIn() {
+    const newDashboardUrl = process.env.NEW_DASHBOARD_URL;
+    if (!newDashboardUrl || !user) return;
+
+    updateUserOptInStatus(NewDashboardOptInStatusEnum.OPTED_IN);
```

Comment:
❓ **question:** is it possible to await for this function to complete? because I'm wondering whether the browser won't cancel the request when it does redirect to a new page